department-of-veterans-affairs/vets-api
# API-33934-v1-error-formatter-update

## Summary

- Adds handling for different formats. 
- Adds tests.

## Related issue(s)

- [API-33934](https://jira.devops.va.gov/browse/API-33934)

## Testing done

- Testing Steps:
-1. Use Postman or Rails C to run the ClaimEstablisher, insert 'invaid' for the [client_key](https://github.com/department-of-veterans-affairs/vets-api/blob/API-33934-v1-error-formatter-update/modules/claims_api/lib/evss_service/base.rb#L73), insert an empty hash for the [form_data](https://github.com/department-of-veterans-affairs/vets-api/blob/API-33934-v1-error-formatter-update/modules/claims_api/app/sidekiq/claims_api/claim_establisher.rb#L13)
2. Place a byebug in [build_error](https://github.com/department-of-veterans-affairs/vets-api/blob/API-33934-v1-error-formatter-update/modules/claims_api/lib/custom_error.rb#L12) to see what error you are getting, and make sure it funnels correctly.
3. Place a byebug in the [rescue](https://github.com/department-of-veterans-affairs/vets-api/blob/API-33934-v1-error-formatter-update/modules/claims_api/app/sidekiq/claims_api/claim_establisher.rb#L49) to make sure the claim saved the status and the evss_response.


## What areas of the site does it impact?
	modified:   modules/claims_api/app/controllers/claims_api/v1/claims_controller.rb
	modified:   modules/claims_api/spec/requests/v1/claims_request_spec.rb

## Acceptance criteria
